src/Settings/Default.hs

```diff
@@ -215,7 +218,7 @@ defaultFlavour = Flavour
     , libraryWays        = defaultLibraryWays
     , rtsWays            = defaultRtsWays
     , splitObjects       = defaultSplitObjects
-    , dynamicGhcPrograms = False
+    , dynamicGhcPrograms = True
```

Comment:
Is this what Make build system does?

Reply:
That is my understanding yes. See [here](https://github.com/ghc/ghc/blob/d36b1ffac9960db70043aaab43c931ce217912ba/mk/config.mk.in#L151):

``` sh
# Use the dynamic way when building programs in the GHC tree. In
# particular, this means that GHCi will use DLLs rather than loading
# object files directly.
ifeq "$(TargetOS_CPP)" "mingw32"
# This doesn't work on Windows yet
DYNAMIC_GHC_PROGRAMS = NO
DYNAMIC_TOO=NO
else ifeq "$(PlatformSupportsSharedLibs)" "NO"
DYNAMIC_GHC_PROGRAMS = NO
else
DYNAMIC_GHC_PROGRAMS = YES
endif
```

Reply:
I checked again and only the cross-compiling flavours turn this off. We do however have to set the value of that field depending on the platform etc, ideally... Any idea?

Reply:
Perhaps, we could switch the type of `dynamicGhcPrograms` from `Bool` to `Action Bool` or even `Predicate`?

Reply:
I implemented `dynamicGhcPrograms :: Action Bool` (because the default implementation is made simpler) in 0ab30d4d3f.